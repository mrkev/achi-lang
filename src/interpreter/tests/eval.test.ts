import { logsFor } from "./logsFor";

test("simple.1", () => {
  const logs = logsFor(`#log "hi"`);
  expect(logs).toMatchSnapshot();
});

test("if.true", () => {
  const logs = logsFor(`
  if (1) {
    #log "pass"
  }
  `);
  expect(logs).toMatchSnapshot();
});

test("if.true", () => {
  const logs = logsFor(`
  if (1) {
    #log "pass"
  }
  `);
  expect(logs).toMatchSnapshot();
});

test("error.nice", () => {
  const logs = logsFor(`

  const one = y;
  #log one
  `);
  expect(logs).toMatchSnapshot();
});

// test("if.false", () => {
//   const logs = logsFor(`
//   if (0) {
//     #log "pass"
//   }
//   `);
//   expect(logs).toEqual([]);
// });

test("class.1", () => {
  const logs = logsFor(`
  class Point(x: number, y: number)
  `);
  expect(logs).toMatchSnapshot();
});

test("func.1", () => {
  const logs = logsFor(`
  function foo matches Card {
    case King(): { }
  };
  #log foo;
  `);
  expect(logs).toMatchSnapshot();
});

test("func.2", () => {
  const logs = logsFor(`
  const foo = (x: number) => { return "hello" };
  #log foo()`);
  expect(logs).toEqual(["hello"]);
});

// test("evaluateExpression.NamedRecordLiteral", () => {
//   // const context = Context.create();
//   // context.values().set("Point");
//   // evaluateExpression();
// });
