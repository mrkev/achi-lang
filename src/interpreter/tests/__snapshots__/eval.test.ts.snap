// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class.1 1`] = `[]`;

exports[`error.nice 1`] = `[Error: Identifier "y" not found (at 2:15)]`;

exports[`func.1 1`] = `
[
  "[Function: foo]",
]
`;

exports[`if.true 1`] = `
[
  "pass",
]
`;

exports[`simple.1 1`] = `
[
  "hi",
]
`;
